json_spec/README.md

```diff
@@ -1,16 +1,2589 @@
+    </tr>
+    <tr>
+        <td><a name="def-teststep"><strong>Test Step</strong></a></td>
+        <td>A test step is a high level container for all the execution artifacts that are emitted by a particular section of a <a href="#def-testrun"><strong>test run</strong></a>. A <a href="#def-testrun"><strong>test run</strong></a> consists of 0-n test steps.</td>
+    </tr>
+    <tr>
+        <td><a name="def-uniqueidentifier"><strong>Unique Identifier</strong></a></td>
+        <td>A string (as defined by JSON data types) that uniquely identifies the message it is in. An entire diagnostic output is considered to be the scope of the uniqueness property.</td>
+    </tr>
+    <tr>
+        <td><a name="def-verdict"><strong>Verdict</strong></a></td>
+        <td>A short pareto-able string indicating the determination of hardware health for a particular <a href="#def-teststep"><strong>test step</strong></a>. A single <a href="#def-diagnosis"><strong>diagnosis</strong></a> can contain one verdict.</td>
+    </tr>
+</table>
+
+
+## Data Types
+
+The following data types are used in this [**specification**](#def-spec) and are constrained by the JSON [specification](https://www.json.org/json-en.html#spec) itself.
+
+<table>
+    <tr>
+        <td><strong>Data Type</strong></td>
+        <td><strong>Description</strong></td>
+    </tr>
+    <tr>
+        <td>Array</td>
+        <td>A list of elements consisting of the same data type.</td>
+    </tr>
+    <tr>
+        <td>Boolean</td>
+        <td>A single bit data type consisting of true or false.</td>
+    </tr>
+    <tr>
+        <td>Null</td>
+        <td>A nullable value</td>
+    </tr>
+    <tr>
+        <td>Number</td>
+        <td>An integer or floating point value expressed in base 10.</td>
+    </tr>
+    <tr>
+        <td>Object</td>
+        <td>A set of key/value pairs. Each key in the key/value pair of an object shall be a unique string.</td>
+    </tr>
+    <tr>
+        <td>String</td>
+        <td>Sequence of characters contained in double quotes. Strings support escape sequences which are outlined in the following section.</td>
+    </tr>
+</table>
+
+This [**specification**](#def-spec) attempts to allow the use of common JSON parsing libraries that are available across multiple languages. This design decision comes with certain concessions that will be discussed below, such as loss of precision for numeric types. We will also address how these limitations of JSON data types can be mitigated. Alternative extensions to JSON such as OData were considered, but these come at the cost of non-standard parsing libraries.
+
+
+### String Escape Sequences
+
+The following escape sequences are supported in the diagnostic output, which are the standard JSON escape sequences. There is no expansion of the JSON specification for escaping.
```

Comment:
nit-pick: After reading this after a long break, the "there is no expansion..." clause seems redundant.

Reply:
done. please check/resolve